.vib/flink/vib-publish.json

```diff
@@ -0,0 +1,81 @@
1qb2JtYW5hZ2VyIGZsaW5rLWpvYm1hbmFnZXI6NjEyMyAtRHJlc3QucG9ydD04MAogICAgICAgICAgICAgICAgICAgIGVjaG8gIjEiID4gL3RtcC9yZWFkeQogICAgICAgICAgICAgICAgICAgIGRvbmUKICAgICAgICAgICAgICAgICAgZmkKICAgICAgICAgICAgICBzdGFydHVwUHJvYmU6CiAgICAgICAgICAgICAgICBleGVjOgogICAgICAgICAgICAgICAgICBjb21tYW5kOgogICAgICAgICAgICAgICAgICAgIC0gc2gKICAgICAgICAgICAgICAgICAgICAtIC1jCiAgICAgICAgICAgICAgICAgICAgLSB8CiAgICAgICAgICAgICAgICAgICAgICBpZiBbICQoY2F0IC90bXAvcmVhZHkpID0gIjEiIF07IHRoZW4KICAgICAgICAgICAgICAgICAgICAgICAgZXhpdCAwCiAgICAgICAgICAgICAgICAgICAgICBlbHNlCiAgICAgICAgICAgICAgICAgICAgICAgIGV4aXQgMQogICAgICAgICAgICAgICAgICAgICAgZmkKICAgICAgICAgICAgICAgIGluaXRpYWxEZWxheVNlY29uZHM6IDQwCiAgICAgICAgICAgICAgICBwZXJpb2RTZWNvbmRzOiAyMAogICAgICAgICAgICAgICAgdGltZW91dFNlY29uZHM6IDEKICAgICAgICAgICAgICAgIGZhaWx1cmVUaHJlc2hvbGQ6IDE1CiAgICAgICAgICAgICAgICBzdWNjZXNzVGhyZXNob2xkOiAxCiAgICAgICAgICAgICAgdm9sdW1lTW91bnRzOgogICAgICAgICAgICAgICAgLSBuYW1lOiBmbGluay1jb25maWctdm9sdW1lCiAgICAgICAgICAgICAgICAgIG1vdW50UGF0aDogL29wdC9iaXRuYW1pL2ZsaW5rL2NvbmYKICAgICAgICAgIHZvbHVtZXM6CiAgICAgICAgICAgIC0gbmFtZTogZmxpbmstY29uZmlnLXZvbHVtZQogICAgICAgICAgICAgIGNvbmZpZ01hcDoKICAgICAgICAgICAgICAgIG5hbWU6IGZsaW5rLWpvYi1jb25maWcKICAgICAgICAgICAgICAgIGl0ZW1zOgogICAgICAgICAgICAgICAgICAtIGtleTogZmxpbmstY29uZi55YW1sCiAgICAgICAgICAgICAgICAgICAgcGF0aDogZmxpbmstY29uZi55YW1sCgogIC0gfAogICAgYXBpVmVyc2lvbjogdjEKICAgIGtpbmQ6IENvbmZpZ01hcAogICAgbWV0YWRhdGE6CiAgICAgIG5hbWU6IGZsaW5rLWpvYi1jb25maWcKICAgICAgbmFtZXNwYWNlOiB7eyBpbmNsdWRlICJjb21tb24ubmFtZXMubmFtZXNwYWNlIiAuIHwgcXVvdGUgfX0KICAgICAgbGFiZWxzOiB7ey0gaW5jbHVkZSAiY29tbW9uLmxhYmVscy5zdGFuZGFyZCIgLiB8IG5pbmRlbnQgNCB9fQogICAgICAgIHt7LSBpZiAuVmFsdWVzLmNvbW1vbkxhYmVscyB9fQogICAgICAgIHt7LSBpbmNsdWRlICJjb21tb24udHBsdmFsdWVzLnJlbmRlciIgKCBkaWN0ICJ2YWx1ZSIgLlZhbHVlcy5jb21tb25MYWJlbHMgImNvbnRleHQiICQgKSB8IG5pbmRlbnQgNCB9fQogICAgICAgIHt7LSBlbmQgfX0KICAgICAgICBhcHAua3ViZXJuZXRlcy5pby9wYXJ0LW9mOiBmbGluawogICAgICAgIGFwcC5rdWJlcm5ldGVzLmlvL2NvbXBvbmVudDogam9ibWFuYWdlcgogICAgICB7ey0gaWYgLlZhbHVlcy5jb21tb25Bbm5vdGF0aW9ucyB9fQogICAgICBhbm5vdGF0aW9uczoge3stIGluY2x1ZGUgImNvbW1vbi50cGx2YWx1ZXMucmVuZGVyIiAoIGRpY3QgInZhbHVlIiAuVmFsdWVzLmNvbW1vbkFubm90YXRpb25zICJjb250ZXh0IiAkICkgfCBuaW5kZW50IDQgfX0KICAgICAge3stIGVuZCB9fQogICAgZGF0YToKICAgICAgZmxpbmstY29uZi55YW1sOiB8KwogICAgICAgIGpvYm1hbmFnZXIucnBjLmFkZHJlc3M6IGZsaW5rLWpvYm1hbmFnZXIKICAgICAgICBqb2JtYW5hZ2VyLnJwYy5wb3J0OiA2MTIzCiAgICAgICAgcmVzdC5hZGRyZXNzOiBmbGluay1qb2JtYW5hZ2VyCiAgICAgICAgcmVzdC5iaW5kLWFkZHJlc3M6IDAuMC4wLjAKICAgICAgICBibG9iLnNlcnZlci5wb3J0OiA2MTI0CiAgICAgICAgcXVlcnkuc2VydmVyLnBvcnQ6IDYxMjUKICAgICAgICByZXN0LnBvcnQ6IDgwCgpqb2JtYW5hZ2VyOgogIGNvbnRhaW5lclBvcnRzOgogICAgcnBjOiA2MTIzCiAgICBodHRwOiA4MDc4CiAgc2VydmljZToKICAgIHR5cGU6IExvYWRCYWxhbmNlcgogICAgcG9ydHM6CiAgICAgIHJwYzogNjEyMwogICAgICBodHRwOiA4MAogIHBvZFNlY3VyaXR5Q29udGV4dDoKICAgIGZzR3JvdXA6IDEwMDIKICBjb250YWluZXJTZWN1cml0eUNvbnRleHQ6CiAgICBydW5Bc1VzZXI6IDEwMDIKdGFza21hbmFnZXI6CiAgc2VydmljZToKICAgIHBvcnRzOgogICAgICBycGM6IDE0MjcyCiAgY29udGFpbmVyUG9ydHM6CiAgICBycGM6IDU3NzkK",
```

Comment:
You are missing the following params:

- `serviceAccount.create` and `serviceAccount.automountServiceAccountToken`, and a GOSS test for them.
- `podSecurityContext.enabled`
- `containerSecurityContext.enabled`

Apart from that, there is no need to include the following properties:

- `taskmanager.containerPorts` --> There are no explicit tests (correct me if I am wrong) that check it and it is not crucial to any of the tests. From, [ref](https://github.com/bitnami/charts/blob/main/TESTING.md#runtime-parameters)

Reply:
PR https://github.com/bitnami/charts/pull/15510 is about to be merged, and it includes changes that deprecate the usage of `base64` params in favor of a more user-friendly text-plain file with the runtime parameter. Please, adapt this PR to follow this improvement 😁

Reply:
There are a lot of properties inside the `extraDeploy` objects that can be removed in order to reduce the length of the whole `runtime_parameters`. For instance:

```diff
extraDeploy:
  - |
    apiVersion: batch/v1
    kind: Job
    metadata:
      name: flink-job-runner
-      namespace: {{ include "common.names.namespace" . | quote }}
-      labels: {{- include "common.labels.standard" . | nindent 4 }}
-       {{- if .Values.commonLabels }}
-        {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
-        {{- end }}
-        app.kubernetes.io/part-of: flink
-      {{- if .Values.commonAnnotations }}
-      annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
-      {{- end }}
    spec:
```

We should try to shrink the size to the minimum

Reply:
Reverted #15510 implementation till merged after talk with @FraPazGal. For the rest, all done!

Reply:
The use of a `runtime-parameters.yaml` file is possible now that #15510 has been merged. Sorry for the noise!

Reply:
Done! Thanks @FraPazGal , amazing feature and contribution! 🔥